scripts/postinstall.js

```diff
@@ -1,13 +1,19 @@
 const { spawn } = require('child_process')
 
-if (process.env.CI) {
-  console.log('Skipping postinstall since CI env variable is set')
-} else {
+function schemas() {
+  if (!process.env.GENERATE_SCHEMAS_ON_INSTALL) process.exit(0)
+  console.log('Generating schemas...')
   const cmd = spawn('yarn schemas', {
     shell: true,
     stdio: 'inherit',
   })
   cmd.on('exit', (code) => {
-    process.exit(code)
+    console.log(`Schema generation completed with exit code ${code}`)
+    if (code !== 0) process.exit(code)
   })
 }
+
+for (const job of [schemas]) {
```

Comment:
Is there a plan to add more postinstall scripts?

Postinstall scripts are considered bad practice and I think we should not add more.

Reply:
Agreed, we don't want them, yet here we are. This at least allows switching specific post-install features one needs individually, instead of running _all_ of post-install.

There are no plans to add more, but I found it strange seeing the generic postinstall hook to only — and always — run schema generation.